velox/functions/prestosql/tests/ArrayNGramTest.cpp

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#include <optional>
+#include "velox/common/base/tests/GTestUtils.h"
+#include "velox/functions/prestosql/tests/utils/FunctionBaseTest.h"
+
+using namespace facebook::velox::test;
+using namespace facebook::velox::functions::test;
+
+namespace {
+
+class ArrayNGramTest : public FunctionBaseTest {
+ protected:
+  template <typename T>
+  void testNgram(
+      const std::vector<std::vector<std::optional<T>>>& inputArray,
+      int64_t size,
+      const std::vector<std::optional<std::vector<std::optional<T>>>>&
+          expectedOutput) {
+    auto input = makeNullableArrayVector<T>({inputArray});
+    auto result =
+        evaluate(fmt::format("ngrams(c0, {})", size), makeRowVector({input}));
+
+    auto expected = makeNullableNestedArrayVector<T>({{expectedOutput}});
+    assertEqualVectors(expected, result);
+  }
+};
+
+TEST_F(ArrayNGramTest, intArrays) {
+  testNgram<int64_t>({{1, 2, 3, 4}}, 1, {{{1}}, {{2}}, {{3}}, {{4}}});
```

Comment:
Could you add some test cases where the input array is NULL. Also, it would be helpful to add test cases where input array is of complex types, e.g., input is array-of-row, array-of-array, or array-of-map vector.